src/pipeline.js

```diff
@@ -94,6 +94,67 @@ class Pipeline {
     this._posts = [];
     // functions that are executed before each step
     this._taps = [];
+
+    this.attach = (ext) => {
+      if (this.sealed) {
+        return;
+      }
+      if (ext && ext.before && typeof ext.before === 'object') {
+        Object.keys(ext.before).map(key => this.attach.before(key, ext.before[key]));
+      }
+      if (ext && ext.after && typeof ext.after === 'object') {
+        Object.keys(ext.after).map(key => this.attach.after(key, ext.after[key]));
+      }
+      this.sealed = true;
+    };
+
+    /**
+     * Registers an extension to the pipeline.
+     * @param {String} name - name of the extension point (typically the function name).
+     * @param {pipelineFunction} f - a new pipeline step that will be injected relative to `name`.
+     * @param {integer} offset - where to insert the new function (0 = before, 1 = after)
```

Comment:
can we have constants for the offset? or do you plan to add real numerical offsets?
why not a boolean?
why not `-1` for _before_?